UndertaleModTool/Converters/UndertaleCachedImageLoader.cs

```diff
@@ -158,42 +156,48 @@ public static void Reset()
             currBufferSize = 1048576;
         }
 
-        public static Bitmap CreateSpriteBitmap(Rectangle rect, in UndertaleTexturePageItem texture, int diffW = 0, int diffH = 0, bool isTile = false)
+        public static SKBitmap CreateSpriteBitmap(Rectangle rect, in UndertaleTexturePageItem texture, int diffW = 0, int diffH = 0, bool isTile = false)
         {
-            using MemoryStream stream = new(texture.TexturePage.TextureData.TextureBlob);
-            Bitmap spriteBMP = new(rect.Width, rect.Height);
+            SKBitmap spriteBMP = new(rect.Width, rect.Height);
 
             rect.Width -= (diffW > 0) ? diffW : 0;
             rect.Height -= (diffH > 0) ? diffH : 0;
             int x = isTile ? texture.TargetX : 0;
             int y = isTile ? texture.TargetY : 0;
 
-            using (Graphics g = Graphics.FromImage(spriteBMP))
+            using (SKCanvas g = new(spriteBMP))
             {
-                using Image img = Image.FromStream(stream); // "ImageConverter.ConvertFrom()" does the same, except it doesn't explicitly dispose MemoryStream
-                g.DrawImage(img, new Rectangle(x, y, rect.Width, rect.Height), rect, GraphicsUnit.Pixel);
+                using SKBitmap img = SKBitmap.Decode(texture.TexturePage.TextureData.TextureBlob);
+                
+                var rectDest = SKRect.Create(x, y, rect.Width, rect.Height);
+                var rectSrc = SKRect.Create(rect.Left, rect.Top, rect.Width, rect.Height);
+                g.DrawBitmap(img, rectSrc, rectDest);
             }
 
             return spriteBMP;
         }
         private ImageSource CreateSpriteSource(in Rectangle rect, in UndertaleTexturePageItem texture, int diffW = 0, int diffH = 0, bool isTile = false)
         {
-            Bitmap spriteBMP = CreateSpriteBitmap(rect, in texture, diffW, diffH, isTile);
+            SKBitmap spriteBMP = CreateSpriteBitmap(rect, in texture, diffW, diffH, isTile);
+            var data = spriteBMP.Encode(SKEncodedImageFormat.Png, 100);
+
+            BitmapImage spriteSrc = new();
```

Comment:
Why cant this be just another SKBitmap?

Reply:
Because `SKBitmap` is not compatible with WPF, while `BitmapImage` (which is `ImageSource`) belongs to it.

Reply:
Ah gotcha.